lib/Controller/MessageApiController.php

```diff
@@ -0,0 +1,171 @@
+		array $cc = [],
+		array $bcc = [],
+		array $references = [],
+	): DataResponse {
+		if($this->userId === null) {
+			return new DataResponse('Account not found.', Http::STATUS_NOT_FOUND);
+		}
+
+		try {
+			$mailAccount = $this->accountService->find($this->userId, $accountId);
+		} catch (ClientException $e) {
+			$this->logger->error("Mail account #$accountId not found", ['exception' => $e]);
+			return new DataResponse('Account not found.', Http::STATUS_NOT_FOUND);
+		}
+
+		if($fromEmail !== $mailAccount->getEmail()) {
+			try {
+				$alias = $this->aliasesService->findByAliasAndUserId($fromEmail, $this->userId);
+			} catch (DoesNotExistException $e) {
+				$this->logger->error("Alias $fromEmail for mail account $accountId not found", ['exception' => $e]);
+				// Cannot send from this email as it is not configured as an alias
+				return new DataResponse("Could not find alias $fromEmail. Please check the logs.", Http::STATUS_NOT_FOUND);
+			}
+		}
+
+		if(empty($to)) {
+			return new DataResponse('Recipients cannot be empty.', Http::STATUS_BAD_REQUEST);
+		}
+
+		$attachments = $this->request->getUploadedFile('attachments');
+		$numberOfAttachments = count($attachments['name'] ?? []);
+
+		$localAttachments = [];
+		$messageAttachments = [];
+		$attachmentErrors = false;
+		while ($numberOfAttachments > 0) {
+			$numberOfAttachments--;
+			$filedata = [
+				'name' => $attachments['name'][$numberOfAttachments],
+				'type' => $attachments['type'][$numberOfAttachments],
+				'size' => $attachments['size'][$numberOfAttachments],
+				'tmp_name' => $attachments['tmp_name'][$numberOfAttachments],
+			];
+			$file = new UploadedFile($filedata);
+			try {
+				$localAttachment = $this->attachmentService->addFile($this->userId, $file);
+				$messageAttachments[] = $localAttachment;
+				$localAttachments[] = ['type' => 'local', 'id' => $localAttachment->getId()];
+			} catch (UploadException $e) {
+				$this->logger->error('Could not convert attachment to local attachment.', ['exception' => $e]);
+				$attachmentErrors = true;
+			}
+		}
+
+		if ($attachmentErrors) {
+			foreach ($localAttachments as $localAttachment) {
+				// Handle possible dangling local attachments
+				$this->attachmentService->deleteAttachment($this->userId, $localAttachment['id']);
+			}
+			return new DataResponse('Could not convert attachment(s) to local attachment(s). Please check the logs.', Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+
+		$message = new LocalMessage();
+		$message->setType(LocalMessage::TYPE_OUTGOING);
+		$message->setAccountId($accountId);
+		$message->setSubject($subject);
+		$message->setBody($body);
+		$message->setEditorBody($body);
+		$message->setHtml($isHtml);
+		$message->setSendAt($this->time->getTime());
+		$message->setType(LocalMessage::TYPE_OUTGOING);
+
+		if(isset($alias)) {
+			$message->setAliasId($alias->getId());
+		}
+		if(!empty($references)) {
+			$message->setInReplyToMessageId($references[0]); // needs DB changes I guess
```

Comment:
what does the comment mean? is there something to fix?

Reply:
Can you leave this information as code comment?

Reply:
Left a comment in the Mail Transmission